src/sinks/databend/event_encoder.rs

```diff
@@ -0,0 +1,20 @@
+use bytes::{BufMut, BytesMut};
+
+use crate::{codecs::Transformer, event::Event};
+
+pub struct DatabendEventEncoder {
+    pub(crate) transformer: Transformer,
+}
+
+impl DatabendEventEncoder {
+    pub(crate) fn encode_event(&mut self, mut event: Event) -> BytesMut {
```

Comment:
Yes! the existing `json` codec can be used. But I'm also considering adding a `csv` format support.
Is it OK to add a `csv` to the existing `lib/codecs`?

Reply:
Yeah, that's been an open feature request https://github.com/vectordotdev/vector/issues/7124 - we'd happy take a PR adding a new encoder for CSV.

Would there be a user-visible difference between encoding the events as `json` vs `csv`, I feel like if it's not impacting the user it may not be worth exposing as a configuration option?